### DISCUSSION TITLE
doris SQL查询貌似完全无法命中rollup。

### DISCUSSION THREAD

**Discussion by w875050772** [apache/doris#26361] [A - General / Q&A]
我在doris中创建表并同时创建了三个rollup，使用以下SQL查询时执行计划未显示有命中rollup

rollup创建情况如下：

`t_erc20_transfer	AGG_KEYS	f_timestamp	LARGEINT	No	true	0		true
		f_contract_address	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_from	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_to	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_tx_hash	VARCHAR(255)	No	true		BLOOM_FILTER	true
		f_log_index_num	VARCHAR(100)	No	true	0		true
		f_height	LARGEINT	No	true	0	BLOOM_FILTER	true
		f_block_hash	VARCHAR(255)	No	false		REPLACE	true
		id	LARGEINT	No	false	0	REPLACE	true
		f_value	VARCHAR(100)	No	false	0	REPLACE	true
		f_is_exp	INT	No	false	0	REPLACE	true
								
r2	AGG_KEYS	f_from	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_contract_address	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_to	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_tx_hash	VARCHAR(255)	No	true		BLOOM_FILTER	true
								
r1	AGG_KEYS	f_tx_hash	VARCHAR(255)	No	true		BLOOM_FILTER	true
		f_from	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_to	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_contract_address	VARCHAR(100)	No	true		BLOOM_FILTER	true
								
r3	AGG_KEYS	f_to	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_contract_address	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_from	VARCHAR(100)	No	true		BLOOM_FILTER	true
		f_tx_hash	VARCHAR(255)	No	true		BLOOM_FILTER	true`
SQL执行计划及结果如下

```
explain select * from t_erc20_transfer t1 where t1.f_contract_address =?
and t1.f_from =?
```

PLAN FRAGMENT 0
  OUTPUT EXPRS:
    `t1`.`f_timestamp`
    `t1`.`f_contract_address`
    `t1`.`f_from`
    `t1`.`f_to`
    `t1`.`f_tx_hash`
    `t1`.`f_log_index_num`
    `t1`.`f_height`
    `t1`.`f_block_hash`
    `t1`.`id`
    `t1`.`f_value`
    `t1`.`f_is_exp`
  PARTITION: HASH_PARTITIONED: `default_cluster:zk`.`t_erc20_transfer`.`f_contract_address`

  VRESULT SINK

  0:VOlapScanNode
     TABLE: default_cluster:zk.t_erc20_transfer(t_erc20_transfer), PREAGGREGATION: OFF. Reason: No AggregateInfo
     PREDICATES: `t1`.`f_contract_address` = '0x000000000000000000000000000000000000800a', `t1`.`f_from` = '0x8216d75737a0b8f6dcc02681e46a724a1b1d0f48'
     partitions=1/1, tablets=1/10, tabletList=9338636
     cardinality=152195999, avgRowSize=294.25452, numNodes=3

在执行计划结果中未见到rollup命中关键字，且查询非常缓慢，请问有什么优化建议

**Comment by ldbing** [comment]
你的是哪个版本的 我的物化视图也无法命中 试了下官方的demo 也不行https://doris.incubator.apache.org/zh-CN/docs/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-MATERIALIZED-VIEW

**Reply by w875050772** [reply]
1.2.1

**Reply by LemonLiTree** [reply]
Can you provide a complete reproduction case?

**Reply by w875050772** [reply]
I've found the cause of the problem, it should be that my result set has fields that aren't defined in the rollup. Thank you.
